loadflow/loadflow-api/src/main/java/com/powsybl/loadflow/LoadFlowProviderUtil.java

```diff
@@ -0,0 +1,52 @@
+package com.powsybl.loadflow;
+
+import com.powsybl.commons.extensions.Extension;
+import com.powsybl.commons.parameters.Parameter;
+import com.powsybl.commons.parameters.ParameterType;
+import com.powsybl.loadflow.json.JsonLoadFlowParameters;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public final class LoadFlowProviderUtil {
+
+    private LoadFlowProviderUtil() {
+
+    }
+
+    private static Object getValueFromString(ParameterType type, String value) {
+        if (value == null || value.isEmpty()) {
+            return type == ParameterType.STRING ? value : null;
+        }
+        return switch (type) {
+            case DOUBLE -> Double.parseDouble(value);
+            case STRING -> value;
+            case BOOLEAN -> Boolean.parseBoolean(value);
+            case INTEGER -> Integer.parseInt(value);
+            case STRING_LIST -> Arrays.asList(value.trim().split("[:,]"));
+        };
+    }
+
+    /**
+     * Retrieves the parameters of the extension associated with this provider,
+     * incorporating any overrides from the loadFlowParamtersLoader.
+     *
+     * @return The parameters of the associated extension with overrides applied from PlatformConfig.
+     */
+    public static List<Parameter> getSpecificParameters(LoadFlowProvider provider, Optional<LoadFlowDefaultParametersLoader> loadFlowDefaultParametersLoader) {
```

Comment:
It is a bad practice to use `Optional` as an input parameter.
The case were the optional is empty should be handle outside this method.